### PR TITLE
Improve and simplify email verification codes generation

### PR DESCRIPTION
**Pull request by zoldar**
### Changes

Verification codes are now generated on the fly, without relying on prepopulated table.

There's a new table holding at most one activation code per user. As there's no risk of depleting codes, cleaning up stale activation code entries is not necessary.

There's also a small fix to seeds which started failing constraint enforcing single owner per site.

~NOTE: This PR will require splitting out migration.~ done

### Tests
- [x] Automated tests have been added
- [ ] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [x] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [x] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode
- [x] This PR does not change the UI

### CODE REVIEW

File: test/plausible_web/controllers/auth_controller_test.exs [hunk 1]
```diff
@@ -203,18 +205,40 @@ defmodule PlausibleWeb.AuthControllerTest do
...
 
-      code =
-        Repo.one(
-          from c in "email_verification_codes",
-            where: c.user_id == ^user.id,
-            select: %{user_id: c.user_id, issued_at: c.issued_at}
+      assert code = Repo.get_by(Auth.EmailActivationCode, user_id: user.id)
+
+      assert code.user_id == user.id
+      assert Timex.after?(code.issued_at, Timex.now() |> Timex.shift(seconds: -10))
```
Comment (aerosol) [inline test/plausible_web/controllers/auth_controller_test.exs:214]: Minor, perhaps export and refute `expired?` here?

File: test/plausible_web/controllers/auth_controller_test.exs [hunk 2]
```diff
@@ -203,18 +205,40 @@ defmodule PlausibleWeb.AuthControllerTest do
...
+
+      assert verification.id == new_verification.id
+      assert verification.user_id == new_verification.user_id
+      # this actually has a chance to fail 1 in 8999 runs
+      # but at the same time it's good to have a confirmation
+      # that it indeed generates a new code
+      if verification.code == new_verification.code do
+        Logger.warn(
+          "Congratulations! You you have hit 1 in 8999 chance of the same " <>
+            "email verification code repeating twice in a row!"
```
Comment (aerosol) [inline test/plausible_web/controllers/auth_controller_test.exs:237]: 😅👏

File: priv/repo/migrations/20231011101825_add_email_activation_codes.exs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+defmodule Plausible.Repo.Migrations.AddEmailActivationCodes do
+  use Ecto.Migration
+
+  def change do
+    create table(:email_activation_codes) do
+      add :code, :integer, null: false
```
Comment (aerosol) [inline priv/repo/migrations/20231011101825_add_email_activation_codes.exs:6]: How about make this a string, to avoid integer/string conversions?

**Review by aerosol — APPROVED** [review]
Beautiful, optional suggestions inline.

File: test/plausible_web/controllers/auth_controller_test.exs (hunk 1 above)
Comment (zoldar) [reply test/plausible_web/controllers/auth_controller_test.exs]: https://github.com/plausible/analytics/pull/3407/commits/3a1b36b329ec0a42131d0f56457093514755659d

File: priv/repo/migrations/20231011101825_add_email_activation_codes.exs (hunk 3 above)
Comment (zoldar) [reply priv/repo/migrations/20231011101825_add_email_activation_codes.exs]: https://github.com/plausible/analytics/pull/3407/commits/7d6cee8f8eaf00f98bb86129afe010b3023f4f40

**Pull request merged**